### DISCUSSION TITLE
Can we bind to documented shoelace component events?

### DISCUSSION THREAD

**Discussion by tkriplean** [hyperdiv/hyperdiv#33] [Help] [answered]
Something like e.g.

```
dialog = hd.dialog("Login", opened = True)
if dialog.sl_hide:   # responding to the sl-hide event from a Dialog
   login_state.show_login_dialog = False
```

Apologies if this already documented, I was not able to find any info on it. 

Thanks,
Travis

**Comment by mariusnita** [comment, marked as answer]
Hi Travis, currently you can do this with:

```py
if dialog.was_closed:
    login_state.show_login_dialog = False
```

`dialog` inherits from [Togglable](https://docs.hyperdiv.io/reference/components/Togglable) which exposes `was_opened` and `was_closed` properties.

**Reply by tkriplean** [reply]
Thanks! Are all of the documented shoelace component events translated somehow into Hyperdiv? If not, is it planned to address them all, or is the idea that Hyperdiv will have Opinions about which events are important to support? 

For my part, I'm not blocked by this at all, but I am trying to navigate the documentation to understand what is supported, and how, as we translate from a hyperdiv environment to the rendered HTML via shoelace etc.

**Reply by mariusnita** [reply]
The translation from Shoelace into Hyperdiv is not 1-to-1, and Hyperdiv indeed makes discretionary decisions about how to surface the underlying Shoelace components to Python. I think going for full 1-to-1 fidelity with Shoelace is unproductive for Hyperdiv. That points to a different kind of framework that gives full-fidelity browser control from Python. I think that's an interesting direction but is not what Hyperdiv is currently going for.

The best reference for Hyperdiv components is the Hyperdiv component documentation. But if you see Shoelace functionality that is glaringly missing from Hyperdiv, we can add it.
